// sequelize-auto/bin/models/event.js
/* jshint indent: 2 */

module.exports = function(sequelize, DataTypes) {
  return sequelize.define('event', {
    event_oid: {
      type: DataTypes.INTEGER(11),
      allowNull: false,
      primaryKey: true,
      autoIncrement: true
    },
    finishingdate: {
      type: DataTypes.DATEONLY,
      allowNull: true
    },
    startingdate: {
      type: DataTypes.DATEONLY,
      allowNull: true
    },
    creationdate: {
      type: DataTypes.DATEONLY,
      allowNull: true
    },
    ispremium: {
      type: DataTypes.BOOLEAN,
      allowNull: true
    },
    description: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    name: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    state_oid: {
      type: DataTypes.INTEGER(11),
      allowNull: true,
      references: {
        model: 'state',
        key: 'oid'
      }
    },
    sport_oid: {
      type: DataTypes.INTEGER(11),
      allowNull: true,
      references: {
        model: 'sport',
        key: 'oid'
      }
    }
  }, {
    tableName: 'event'
  });
};
